### ISSUE TITLE
Declaring Symbol variable conflicts with namespaceToStringTag

### ISSUE THREAD

**Issue by yyx990803** [rollup/rollup#3969]
- Rollup Version: 2.39.0
- Operating System (or Browser): MacOS
- Node Version (if applicable):
- Link to reproduction (IMPORTANT, read below): https://github.com/yyx990803/rollup-tostringtag-symbol-conflict

The bug occurs when combining the following:

1. `-f cjs`
2. [--namespaceToStringTag](https://rollupjs.org/guide/en/#outputnamespacetostringtag)
3. Code: `var Symbol = ...`

### Expected Behavior

Generated code should work.

### Actual Behavior

The generated code:

```js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });
exports[Symbol.toStringTag] = 'Module';

var Symbol = 1;

exports.Symbol = Symbol;
```

Due to the `var` hoisting behavior, `Symbol` is `undefined` then `exports[Symbol.toStringTag] = 'Module';` is executed.

This my seem like a very niche case but the actual code was found in `lodash-es`. i.e. if the user tries to bundle `lodash-es` with the above options they will run into the same error.

### Possible Fixes

Make `Symbol` a reserved variable name (i.e. always consider it already declared) so that `var Symbol` in user code is rewritten to `var Symbol$1` or something similar.

**Comment by lukastaegert** [comment]
I see, that is not good. Will need to check if this conflicts in any way with how Polyfills handle Symbol.

**Comment by lukastaegert** [comment]
Of course there are no issues with Polyfills as they all appear to first identify the global object and operate on that. Fix at #3971. This also avoids the same problem for `Object` and `Promise`, which Rollup also relies on for various things.
